APIKit/RequestObserverType.swift

```diff
@@ -0,0 +1,7 @@
+import Foundation
+import Result
+
+public protocol RequestObserverType {
```

Comment:
How about adding empty default implementations? That enables users to implement (override) only one handler they need.